### PR TITLE
StateProofs: Spv tracking cr fixes

### PR DESCRIPTION
**Pull request by almog-t**

### CODE REVIEW

File: ledger/accountdb.go [hunk 1]
```diff
@@ -5315,23 +5315,23 @@ func deleteUnfinishedCatchpoint(ctx context.Context, e db.Executable, round basi
...
 
-func insertStateProofVerificationData(ctx context.Context, tx *sql.Tx, data []verificationCommitData) error {
-	if len(data) == 0 {
+func insertStateProofVerificationContext(ctx context.Context, tx *sql.Tx, contexts []verificationCommitContext) error {
+	if len(contexts) == 0 {
 		return nil
 	}
 
-	insertStmt, err := tx.PrepareContext(ctx, "INSERT INTO stateproofverification(targetstateproofround, verificationdata) VALUES(?, ?)")
+	insertStmt, err := tx.PrepareContext(ctx, "INSERT INTO stateproofverification(targetstateproofround, verificationcontext) VALUES(?, ?)")
```
Comment (algoidan) [inline ledger/accountdb.go:5323]: should we change `targetstateproofround` -> to `lastattestedround`?

File: ledger/accountdb.go [hunk 2]
```diff
@@ -5344,7 +5344,7 @@ func insertStateProofVerificationData(ctx context.Context, tx *sql.Tx, data []ve
 	return nil
 }
 
-func deleteOldStateProofVerificationData(ctx context.Context, tx *sql.Tx, earliestTrackStateProofRound basics.Round) error {
+func deleteOldStateProofVerificationContext(ctx context.Context, tx *sql.Tx, earliestTrackStateProofRound basics.Round) error {
```
Comment (algoidan) [inline ledger/accountdb.go:5347]: maybe `earliestTrackStateProofRound` -> `earliestLastAttestedRound`?

File: ledger/stateproofverificationtracker.go [hunk 3]
```diff
@@ -159,122 +159,123 @@ func (spt *stateProofVerificationTracker) handleUnorderedCommit(*deferredCommitC
...
+		err = fmt.Errorf("%w for round %d: %s", errStateProofVerificationContextNotFound, stateProofLastAttestedRound, err)
 	}
 
-	return verificationData, err
+	return verificationContext, err
 }
 
-func (spt *stateProofVerificationTracker) committedRoundToLatestCommitDataIndex(committedRound basics.Round) int {
-	latestCommittedDataIndex := -1
+func (spt *stateProofVerificationTracker) committedRoundToLatestCommitContextIndex(committedRound basics.Round) int {
```
Comment (algoidan) [inline ledger/stateproofverificationtracker.go:208]: maybe `committedRoundToLatestCommitContextIndex` -> `roundToLatestCommitContextIndex`?

File: ledger/tracker.go [hunk 4]
```diff
@@ -271,12 +271,14 @@ type deferredCommitContext struct {
 	stats       telemetryspec.AccountsUpdateMetrics
 	updateStats bool
 
-	// state proof verification deletion information
-	stateProofVerificationLatestDeleteDataIndex        int
-	stateProofVerificationEarliestTrackStateProofRound basics.Round
+	spVerification struct {
+		// state proof verification deletion information
+		LatestUsedDeleteContextIndex int
+		EarliestTrackStateProofRound basics.Round
```
Comment (algoidan) [inline ledger/tracker.go:277]: `EarliestTrackStateProofRound` -> `EarliestLastAttestedRound`

File: ledger/stateproofverificationtracker.go [hunk 5]
```diff
@@ -118,40 +118,40 @@ func (spt *stateProofVerificationTracker) produceCommittingTask(_ basics.Round,
...
+	lastContextToCommitIndex := spt.roundToLatestCommitContextIndex(dcc.newBase)
+	dcc.spVerification.CommitContext = make([]verificationCommitContext, lastContextToCommitIndex+1)
+	copy(dcc.spVerification.CommitContext, spt.trackedCommitContext[:lastContextToCommitIndex+1])
 
-	dcc.stateProofVerificationLatestDeleteDataIndex = spt.committedRoundToLatestDeleteDataIndex(dcc.newBase)
-	if dcc.stateProofVerificationLatestDeleteDataIndex > -1 {
-		dcc.stateProofVerificationEarliestTrackStateProofRound = spt.trackedDeleteData[dcc.stateProofVerificationLatestDeleteDataIndex].stateProofNextRound
+	dcc.spVerification.LatestUsedDeleteContextIndex = spt.roundToLatestDeleteContextIndex(dcc.newBase)
+	if dcc.spVerification.LatestUsedDeleteContextIndex >= 0 {
+		dcc.spVerification.EarliestTrackLastAttestedRound = spt.trackedDeleteContext[dcc.spVerification.LatestUsedDeleteContextIndex].stateProofNextRound
```
Comment (algorandskiy) [inline ledger/stateproofverificationtracker.go:130]: I still think these names should be shorter

**Review by algorandskiy — APPROVED** [review]

File: ledger/stateproofverificationtracker.go (hunk 5 above)
Comment (almog-t) [reply ledger/stateproofverificationtracker.go]: Shortened some more.

**Review by algoidan — APPROVED** [review]

**Pull request merged**